llvm/lib/Target/CheerpBackend/CheerpTargetTransformInfo.cpp

```diff
@@ -0,0 +1,111 @@
+//===-- CheerpTargetTransformInfo.cpp - Cheerp helper --------------------===//
+//
+//                     Cheerp: The C++ compiler for the Web
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+// Copyright 2022 Leaning Technologies
+//
+//===----------------------------------------------------------------------===//
+
+#include "CheerpTargetTransformInfo.h"
+#include "llvm/CodeGen/CostTable.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Cheerp/CommandLine.h"
+using namespace llvm;
+
+#define DEBUG_TYPE "wasmtti"
+
+unsigned CheerpTTIImpl::getNumberOfRegisters(unsigned ClassID) const {
```

Comment:
I am a bit confused by the logic of this function. what does it return for scalars if simd is disabled? it seems 0 but isn't that wrong?

Reply:
I should probably redo the logic here, put all the if statements inside the if(Vector) block.

Reply:
Done